ios/engine/KMEI/KeymanEngine/Classes/Resource Management/ResourceDownloadQueue.swift

```diff
@@ -107,6 +119,26 @@ class DownloadBatch<Resource: LanguageResource, Package: TypedKeymanPackage<Reso
       return resArray
     }
   }
+
+  public func completeWithCancellation() {
+    completionBlock?(nil, nil)
+  }
+
+  public func completeWithError(error: Error) {
+    completionBlock?(nil, error)
+  }
+
+  public func completeWithPackage(fromKMP file: URL) {
+    do {
+      if let package = try ResourceFileManager.shared.prepareKMPInstall(from: file) as? Resource.Package {
+        completionBlock?(package, nil)
+      } else {
+        completionBlock?(nil, KMPError.invalidPackage)
+      }
+    } catch {
+      completionBlock?(nil, error)
+    }
+  }
```

Comment:
This set of methods (on `DownloadBatch`) here is probably the highlight of this PR.

Once we fully convert iOS over to downloading KMPs, the core code for `ResourceDownloadQueue` won't even need to know what type of `LanguageResource` it's working with - even to return the Package.  The "batch" will produce the correctly-typed package for the completion callbacks instead of the queue.